test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -1143,6 +1144,30 @@ protected static void wipeAllIndices(boolean preserveSecurityIndices) throws IOE
         }
     }
 
+    // Make warnings handler that ignores the .async-search warning since .async-search may randomly appear when async requests are slow
```

Comment:
```suggestion
    // Make warnings handler ignore the .async-search warning since .async-search may randomly appear when async requests are slow
```

Reply:
Shouldn't we wait for those async requests then maybe? Is the index randomly popping up because we have those complete in the background somehow?

Reply:
I think we should do both things. We should cancel any outstanding async requests and wait for them to die. I think folks have mostly been doing that in "after" clauses, but I feel like that's something we'd traditionally handle in the teardown code here.

But if the async-search index exists after the test we should trash it so we don't leak any async results from one test to another.

Reply:
I've made https://github.com/elastic/elasticsearch/issues/117376 so we don't lose track of this concern.